sdks/java/io/elasticsearch-tests/elasticsearch-tests-common/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOTestCommon.java

```diff
@@ -1229,4 +1230,16 @@ void testWriteWithIsDeletedFnWithoutPartialUpdate() throws Exception {
     assertEquals(numDocs / 2, currentNumDocs);
     assertEquals(0, countByScientistName(connectionConfiguration, restClient, "Darwin", null));
   }
+
+  void testValidSSLAndUsernameConfiguration(String filePath) throws Exception {
+    ConnectionConfiguration configWithSsl =
+        connectionConfiguration
+            .withUsername("username")
+            .withPassword("password")
+            .withKeystorePassword("qwerty")
+            .withKeystorePath(filePath);
+
+    RestClient restClient = configWithSsl.createClient();
+    Assert.assertNotNull("rest client should not be null", restClient);
```

Comment:
can you add assertions to verify that the `setHttpClientConfigCallback` is as expected? I think that assertion is one that would have failed prior to your changes, and will pass now i.e. validates that your change fixes a preexisting issue.  After that this looks good to merge!